lua/entities/cfc_simple_ent_antigrav_grenade.lua

```diff
@@ -0,0 +1,172 @@
+AddCSLuaFile()
+
+DEFINE_BASECLASS( "cfc_simple_ent_bubble_grenade" )
+
+ENT.Base = "cfc_simple_ent_bubble_grenade"
+
+ENT.Model = Model( "models/weapons/w_eq_fraggrenade.mdl" )
+
+ENT.BubbleRadius = 250
+ENT.BubbleDuration = 7
+ENT.BubbleGrowDuration = 0.25
+ENT.BubbleShrinkDuration = 0.25
+ENT.EffectLingerOutsideBubble = 0
+
+ENT.GravityMult = -2.5
+ENT.PushStrength = 260 -- Pushes the player up to get them off the ground.
+ENT.FuseOnImpact = 1 -- On the first impact, shortens the remaining fuse time to this.
+
+
+function ENT:SetTimer( delay )
+    BaseClass.SetTimer( self, delay )
+
+    self.Beep = CurTime()
+end
+
+function ENT:Initialize()
+    BaseClass.Initialize( self )
+
+    self:SetMaterial( "models/weapons/w_models/cfc_frag_grenade/frag_grenade_antigrav" )
+
+    local fuseOnImpact = self.FuseOnImpact
+
+    if SERVER and fuseOnImpact then
+        timer.Simple( 0, function()
+            if not IsValid( self ) then return end
+
+            local fuseShortened = false
+
+            function self:PhysicsCollide()
+                local fuseLeft = self._explodeDelay
+                if not fuseLeft then return end
+                if fuseShortened then return end
+
+                fuseShortened = true
+
+                if fuseLeft > fuseOnImpact then
+                    self:SetTimer( fuseOnImpact )
+                end
+            end
+        end )
+    end
```

Comment:
Does this actually need a timer?
```suggestion
    if not SERVER then return end
    if not fuseOnImpact then return end

    timer.Simple( 0, function()
        if not IsValid( self ) then return end

        local fuseShortened = false

        function self:PhysicsCollide()
            local fuseLeft = self._explodeDelay
            if not fuseLeft then return end
            if fuseShortened then return end

            fuseShortened = true

            if fuseLeft > fuseOnImpact then
                self:SetTimer( fuseOnImpact )
            end
        end
    end )
```

Reply:
Unfortunately, yes. Without the timer, it instantly triggers the physcollide upon spawning.

Reply:
Could you add a note in there explaining tha